java/org/apache/catalina/filters/CsrfPreventionFilter.java

```diff
@@ -53,6 +58,25 @@ public class CsrfPreventionFilter extends CsrfPreventionFilterBase {
 
     private String nonceRequestParameterName = Constants.CSRF_NONCE_REQUEST_PARAM;
 
+    private boolean enforce = true;
+
+    private Collection<Predicate<String>> noNoncePatterns = DEFAULT_NO_NONCE_URL_PATTERNS;
+
+    private static final Collection<Predicate<String>> DEFAULT_NO_NONCE_URL_PATTERNS;
+
+    static {
+        ArrayList<Predicate<String>> defaultNoNonceURLPatterns = new ArrayList<>();
+
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".css"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".js"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".gif"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".png"));
+        defaultNoNonceURLPatterns.add(new SuffixPredicate(".jpg"));
```

Comment:
`jpeg`, `jpg` is a DOS problem.

Reply:
I don't follow. How is allowing requests for .jpg and .jpeg (not mentioned here) a DOS problem? Feel free to message me privately if something should be kept quiet.

Oh... did you mean Disk Operating System or Denial of Service?

Reply:
The original extension is `.jpeg`. `.jpg` only exists because IBM's Disk Operating System as too stupid to handle more than three chars for file extensions.

Reply:
I've used UNIX-like systems for decades and I rarely see `.jpeg`, but I'm happy to add it. I was trying to keep the out-of-the-box processing to a minimum.

Reply:
I've added `.jpeg`

Reply:
"I never type `.jpeg` because I can get the same money for `.jpg`" - Mark Twain

But sure, it doesn't hurt to add it as others do like the longhand (is that a word? opposite of shorthand) form